crates/bindings-csharp/Runtime/Runtime.cs

```diff
@@ -77,6 +59,14 @@ public AlgebraicType GetAlgebraicType(ITypeRegistrar registrar) =>
     }
 }
 
+public abstract class BaseReducerContext<DbView> : DbContext<DbView>, IReducerContext
+    where DbView : class, new()
+{
+    public Identity Sender => Runtime.SenderIdentity!;
```

Comment:
Why are these set on the singleton now? What happens if someone wants to store a `ReducerContext`? It would now lose its own identity and point to the one from the currently running reducer instead.

I think those should be own fields, like before.

Reply:
Reducer context is created once for the duration of the module, and @gefjon says each reducer invocation should be seen as its own isolated instance with no memory able to outlive that invocation.

Making them fields would need a new ReducerContext per reducer invocation, which means its DbView instance has to either also be created per invocation, or stored somewhere.

Reply:
I'm fine with either approaches; I guess it depends on what semantics we want for memory across reducer invocations, and if we want to leverage isolation in the impl like this ReducerContext does here.

Reply:
> Making them fields would need a new ReducerContext per reducer invocation

Yeah, that's what I'm suggesting and what Rust is doing as well. Even if users shouldn't rely on memory sharing, we still should avoid singletons to prevent accidentally leaking data between different invocations that might belong to different identities.

Reply:
It appears we could also merge `BaseReducerContext` and `DbContext` into one thing, since they always should be the same?

Reply:
(I accidentally marked this resolved, but unresolved back because the earlier singleton comment is still valid)